androidplot-core/src/main/java/com/androidplot/util/FastNumber.java

```diff
@@ -64,6 +64,32 @@ public double doubleValue() {
         return doublePrimitive;
     }
 
+    /**
+     * To be equal, two instances must both be instances of {@link FastNumber}. The inner {@link
+     * #number} field must also be a common type. Numbers which are mathematically equal are not
+     * necessarily equal. This keeps with the java implementation of common Number classes where for
+     * instance {@code new Integer(0).equals(new Double(0))} returns {@code false}
+     */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        FastNumber that = (FastNumber) o;
+
+        return number != null ? number.equals(that.number) : that.number == null;
+
+    }
+
+    @Override
+    public int hashCode() {
+        return number != null ? number.hashCode() : 0;
```

Comment:
This would result in the following being true:
`new Integer(0).hashCode() == new FastNumber(null).hashCode()`

Since FastNumber is meant to be pseudo-immutable, a FastNumber(null) really doesnt make sense.  I think we should mark the number param in the FastNumber constructor as `\@NonNull` and maybe add a null check with an explicit `IllegalArgumentException` for good measure.  Then we can just `return number.hashCode()`.

It also looks like `FixedSizeEditableXYSeries` instantiates `FastNumber` in a couple places without null checking.  Those would need to be fixed too.